scripts/packaging/unitypackage.ps1

```diff
@@ -205,24 +161,34 @@ foreach ($entry in $packages.GetEnumerator()) {
     # The exportPackages flag expects the last value in the array
     # to be the final output destination.
     $exportPackages = $folders + $unityPackagePath
+    $exportPackages = $exportPackages -join " "
 
     Write-Verbose "Generating .unitypackage: $unityPackagePath"
     Write-Verbose "Log location: $logFileName"
     
     try {
-        $unityArgs = @{
-            'BatchMode'       = $true
-            'AcceptAPIUpdate' = $true
-            'Quit'            = $true
-            'PassThru'        = $true
-            'Wait'            = $true
-            'ExportPackage'   = $exportPackages
-            'LogFile'         = $logFileName
-            'Version'         = $UnityVersion
+        $unityArgs = "-BatchMode -Quit -Wait " +
+            "-projectPath $RepoDirectory " +
+            "-exportPackage $exportPackages " +
+            "-logFile $logFileName"
+
+        # Starts the Unity process, and the waits (and shows output from the editor in the console
+        # while the process is still running.)
+        $proc = Start-Process -FilePath "$unityEditor" -ArgumentList "$unityArgs" -PassThru
+        $ljob = Start-Job -ScriptBlock { param($log) Get-Content "$log" -Wait } -ArgumentList $logFileName
+
+        while (-not $proc.HasExited -and $ljob.HasMoreData)
+        {
+            Receive-Job $ljob
+            Start-Sleep -Milliseconds 200
         }
+        Receive-Job $ljob
+        Stop-Job $ljob
+        Remove-Job $ljob
+
+        Stop-Process $proc
 
-        $process = Start-UnityEditor @unityArgs
-        if (($process.ExitCode -eq 0) -and (Test-Path $unityPackagePath)) {
+        if (($proc.ExitCode -eq 0) -and (Test-Path $unityPackagePath)) {
             Write-Verbose "Successfully created $unityPackagePath"
         }
         else { Write-Error "Failed to create $unityPackagePath" }
```

Comment:
The script should explicitly set the error code ($rc = ... followed by exit $rc at the end of the script).
If unset, the RC will equal to the error code from the last command which doesn't really work if there are some clean up steps involved).

Reply:
doh, I missed this update. I'll do what we did in unitybuild.yml